src/expr/src/scalar/func.rs

```diff
@@ -2253,6 +2254,7 @@ impl BinaryFunc {
             BinaryFunc::ModFloat32 => eager!(mod_float32),
             BinaryFunc::ModFloat64 => eager!(mod_float64),
             BinaryFunc::ModNumeric => eager!(mod_numeric),
+            BinaryFunc::ValueEq => Ok(eager!(eq)),
```

Comment:
How come both `ValueEq` and `Eq` evaluate with `eq`?

Reply:
Yes, the difference is in `propagates_null` and how `eager` macro uses it:

```
        macro_rules! eager {
            ($func:expr $(, $args:expr)*) => {{
                let a = a_expr.eval(datums, temp_storage)?;
                let b = b_expr.eval(datums, temp_storage)?;
                if self.propagates_nulls() && (a.is_null() || b.is_null()) {
                    return Ok(Datum::Null);
                }
                $func(a, b $(, $args)*)
            }}
        }
```

For `ValueEq`, `propagates_null` is false, so we always call the `eq` function. Whereas `Eq` does propagate nulls.